ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blobs/versions/deneb/BlobSidecar.java

```diff
@@ -39,6 +39,9 @@ public class BlobSidecar
         SignedBeaconBlockHeader,
         SszBytes32Vector> {
 
+  private volatile boolean kzgAndInclusionProofValidated = false;
+  private volatile boolean signatureValidated = false;
```

Comment:
I find these validation step a bit confusing and I'm wondering if we can't simply mark the blob sidecar as valid or not (merge them) or split the `kzgAndInclusionProofValidated` into two validation flags: `kzgValidated` and `inclusionProofValidated`. Or do we really need this design for some reason?!

Reply:
the reason I put kzg and inclusion proof together and signatureValidation as a separate boolean is because we can perform kzg and inclusion proof validation just having the blob itself (you don't need anything else). For the signature validation you need a prevalidated signed block OR the state to get the pubkey from and do the signature check.